File: dehaze-java/src/main/java/com/pei/dehaze/model/entity/SysAiSchedule.java
package com.pei.dehaze.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import com.pei.dehaze.common.base.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.io.Serial;
import java.time.LocalDateTime;
import java.util.Map;

/**
 * AI 定时任务
 *
 * @author dehaze
 */
@Data
@EqualsAndHashCode(callSuper = false)
@TableName(value = "sys_ai_schedule", autoResultMap = true)
public class SysAiSchedule extends BaseEntity {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private String name;

    private String cron;

    private String timezone;

    /** input/output 是 SQL 关键字，必须带反引号，否则 MyBatis-Plus 自动生成的列清单会被 JSQLParser 解析失败 */
    @TableField(value = "`input`", typeHandler = JacksonTypeHandler.class)
    private Map<String, Object> input;

    @TableField(value = "`output`", typeHandler = JacksonTypeHandler.class)
    private Map<String, Object> output;

    private Integer enabled;

    private Integer status;

    private Integer circuitStreak;

    private LocalDateTime nextTriggerTime;

    @TableLogic(value = "0", delval = "id")
    private Long deleted;

    @Serial
    private static final long serialVersionUID = 1L;
}

File: dehaze-go/internal/model/sys_ai_schedule.go
package model

import "time"

// SysAiSchedule AI 对话定时任务
type SysAiSchedule struct {
	ID              int64      `gorm:"column:id;primaryKey;autoIncrement" json:"id"`
	UserID          int64      `gorm:"column:user_id" json:"userId"`
	Name            string     `gorm:"column:name" json:"name"`
	Cron            string     `gorm:"column:cron" json:"cron"`
	Timezone        string     `gorm:"column:timezone" json:"timezone"`
	Input           string     `gorm:"column:input;type:json;default:null" json:"input"`
	Output          string     `gorm:"column:output;type:json;default:null" json:"output"`
	Enabled         int        `gorm:"column:enabled" json:"enabled"`
	Status          int        `gorm:"column:status" json:"status"`
	CircuitStreak   int        `gorm:"column:circuit_streak" json:"circuitStreak"`
	NextTriggerTime *time.Time `gorm:"column:next_trigger_time" json:"nextTriggerTime"`
	Deleted         int64      `gorm:"column:deleted" json:"deleted"`
	CreateBy        *int64     `gorm:"column:create_by" json:"createBy"`
	UpdateBy        *int64     `gorm:"column:update_by" json:"updateBy"`
	CreateTime      time.Time  `gorm:"column:create_time;autoCreateTime" json:"createTime"`
	UpdateTime      *time.Time `gorm:"column:update_time;autoUpdateTime" json:"updateTime"`
}

func (SysAiSchedule) TableName() string { return "sys_ai_schedule" }

// SysAiScheduleRun 定时任务执行历史（只追加）
type SysAiScheduleRun struct {
	ID             int64     `gorm:"column:id;primaryKey;autoIncrement" json:"id"`
	ScheduleID     int64     `gorm:"column:schedule_id" json:"scheduleId"`
	UserID         int64     `gorm:"column:user_id" json:"userId"`
	WindowStart    time.Time `gorm:"column:window_start" json:"windowStart"`
	Status         int       `gorm:"column:status" json:"status"`
	SkipReason     *string   `gorm:"column:skip_reason" json:"skipReason"`
	Credits        *float64  `gorm:"column:credits" json:"credits"`
	DurationMs     *int      `gorm:"column:duration_ms" json:"durationMs"`
	ErrorMsg       *string   `gorm:"column:error_msg" json:"errorMsg"`
	ConversationID *int64    `gorm:"column:conversation_id" json:"conversationId"`
	RequestID      *string   `gorm:"column:request_id" json:"requestId"`
	CreateTime     time.Time `gorm:"column:create_time;autoCreateTime" json:"createTime"`
}

func (SysAiScheduleRun) TableName() string { return "sys_ai_schedule_run" }
